<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>3112</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CS 123: Introduction to Computer Graphics</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::ForceTabbedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_10">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="Canvas2D" name="canvas2D">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>456</width>
         <height>3088</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="filterDock">
   <property name="windowTitle">
    <string>Filter</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="filterDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QGroupBox" name="filterType">
       <property name="title">
        <string>Filter Type</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QRadioButton" name="filterTypeInvert">
          <property name="text">
           <string>Invert</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="filterTypeGrayscale">
          <property name="text">
           <string>Grayscale</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="filterTypeIdentity">
          <property name="text">
           <string>Identity</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="filterTypeShiftLeft">
          <property name="text">
           <string>Shift Left</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="filterTypeShiftRight">
          <property name="text">
           <string>Shift Right</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="filterButton">
       <property name="text">
        <string>Filter!</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="filterDockEmptySpace" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionRevert"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuToolbars">
    <property name="title">
     <string>&amp;Toolbars</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuToolbars"/>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRevert">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Revert Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionCopy3Dto2D">
   <property name="text">
    <string>Copy 3D tab to 2D tab</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear Image</string>
   </property>
   <property name="toolTip">
    <string>Clear Image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionUseOrbitingCamera">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use Orbiting Camera</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Canvas2D</class>
   <extends>QWidget</extends>
   <header>Canvas2D.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>filterImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>1573</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRevert</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>revertImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>1555</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>checkAllRayFeatures()</slot>
  <slot>uncheckAllRayFeatures()</slot>
  <slot>setAspectRatioText()</slot>
  <slot>fileNew()</slot>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>renderImage()</slot>
  <slot>cancelRender()</slot>
  <slot>fileCopy3Dto2D()</slot>
  <slot>setCameraAxisX()</slot>
  <slot>setCameraAxisY()</slot>
  <slot>setCameraAxisZ()</slot>
  <slot>updateCameraTranslation()</slot>
  <slot>updateCameraRotationN()</slot>
  <slot>updateCameraRotationV()</slot>
  <slot>updateCameraRotationU()</slot>
  <slot>resetUpVector()</slot>
  <slot>updateCameraClip()</slot>
  <slot>updateCameraHeightAngle()</slot>
  <slot>setCameraAxonometric()</slot>
  <slot>clearImage()</slot>
  <slot>revertImage()</slot>
  <slot>resetSliders()</slot>
 </slots>
</ui>
